app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -186,6 +212,47 @@ open class ProfilesViewModel(
         onFailure = { e -> handleError(e) })
   }
 
+  /**
+   * Converts an image URI to a JPEG byte array, cropping the image to a square at the center.
+   *
+   * @param context The context used to access the content resolver.
+   * @param imageUri The URI of the image to be converted.
+   * @param quality The quality of the JPEG compression (0-100). Default is 100.
+   * @return A byte array representing the JPEG image, or null if an error occurs.
+   */
+  private fun imageUriToJpgByteArray(
```

Comment:
One security issue seems to appear : if the user uploads an image that is very large and heavy, it would still be uploaded to the database.
I think the image should be reduced to a fixed size (like 600x600) to prevent this.

Reply:
You are absolutely right. I don't think this is a security problem but it definitely does not help with the slow upload speed. This is now fixed